src/MainWindow.vala

```diff
@@ -110,7 +111,11 @@ public class Camera.MainWindow : Gtk.ApplicationWindow {
         Idle.add (() => {
             GenericArray<ClutterGst.CameraDevice> camera_devices = camera_manager.get_camera_devices ();
 
-            if (camera_devices.length > 0 && camera_devices[0].get_name () != null) {
+            if (camera_devices[0].get_name () != null) {
```

Comment:
I suppose we can lessen the scope of `camera_name` and do a DRY here:

```diff
diff --git a/src/MainWindow.vala b/src/MainWindow.vala
index 72869bf..c608f3c 100644
--- a/src/MainWindow.vala
+++ b/src/MainWindow.vala
@@ -32,7 +32,6 @@ public class Camera.MainWindow : Gtk.ApplicationWindow {
     };
 
     private uint configure_id;
-    private string camera_name = _("camera");
 
     private Gtk.Stack stack;
     private Granite.Widgets.AlertView no_device_view;
@@ -111,12 +110,10 @@ public class Camera.MainWindow : Gtk.ApplicationWindow {
         Idle.add (() => {
             GenericArray<ClutterGst.CameraDevice> camera_devices = camera_manager.get_camera_devices ();
 
-            if (camera_devices[0].get_name () != null) {
-                camera_name = camera_devices[0].get_name ();
-            }
+            string camera_name = camera_devices[0].get_name () ?? "camera";
 
             if (camera_devices.length > 0) {
-                initialize_camera_view ();
+                initialize_camera_view (camera_name);
             } else {
                 stack.set_visible_child_name ("no-device");
 
@@ -127,7 +124,7 @@ public class Camera.MainWindow : Gtk.ApplicationWindow {
         });
     }
 
-    private void initialize_camera_view () {
+    private void initialize_camera_view (string camera_name) {
         camera_view = new Widgets.CameraView ();
         camera_view.get_camera_device ().set_capture_resolution (640, 480);
 ```

Reply:
indeed, that'd be better, you can even use `unowned string camera_name = camera_devices[0].get_name () ?? "camera";`